modules/admin-ui-picard/app/src/thunks/eventThunks.js

```diff
@@ -232,5 +314,202 @@ export const postNewEvent = async (values, metadataInfo) => {
+            location: data[i].agentConfiguration['event.location'],
+            changedLocation: data[i].agentConfiguration['event.location'],
+            deviceInputs: data[i].agentConfiguration['capture.device.names'],
+            changedDeviceInputs: [],
+            startTimeHour: makeTwoDigest(startDate.getHours()),
+            changedStartTimeHour: makeTwoDigest(startDate.getHours()),
+            startTimeMinutes: makeTwoDigest(startDate.getMinutes()),
+            changedStartTimeMinutes: makeTwoDigest(startDate.getMinutes()),
+            endTimeHour: makeTwoDigest(endDate.getHours()),
+            changedEndTimeHour: makeTwoDigest(endDate.getHours()),
+            endTimeMinutes: makeTwoDigest(endDate.getMinutes()),
+            changedEndTimeMinutes: makeTwoDigest(endDate.getMinutes()),
+            weekday: weekdays[startDate.getDay()].name,
+            changedWeekday: weekdays[startDate.getDay()].name
+        }
+        editedEvents.push(event);
+    }
+
+    return editedEvents;
+};
+
+// check if there are any scheduling conflicts with other events
+export const checkForSchedulingConflicts = events =>  async dispatch => {
+    const formData = new FormData();
+    let update = [];
+    let timezone = moment.tz.guess();
+    for (let i = 0; i < events.length; i++) {
+        update.push({
+            events: [events[i].eventId],
+            scheduling: {
+                timezone: timezone,
+                start: {
+                    hour: parseInt(events[i].changedStartTimeHour),
+                    minute: parseInt(events[i].changedStartTimeMinutes)
+                },
+                end: {
+                    hour: parseInt(events[i].changedEndTimeHour),
+                    minutes: parseInt(events[i].changedEndTimeMinutes)
+                },
+                weekday: events[i].changedWeekday,
+                agentId: events[i].changedLocation
+            }
+        });
+    }
+
+    formData.append('update', JSON.stringify(update));
+
+    let response = [];
+
+    axios.post('/admin-ng/event/bulk/conflicts', formData)
+        .then(res => console.log(res))
+        .catch(res => {
+            if (res.status === 409) {
+                dispatch(addNotification('error', 'CONFLICT_BULK_DETECTED', -1, null, NOTIFICATION_CONTEXT));
+                response = res.data;
+            }
+            console.log(res);
+        });
+
+    return response;
+};
+
+// update multiple scheduled events at once
+export const updateScheduledEventsBulk = values => async dispatch => {
+    let formData = new FormData();
+    let update = [];
+    let timezone = moment.tz.guess();
+
+    for (let i = 0; i < values.changedEvents.length; i++) {
+        let eventChanges = values.editedEvents.find(event => event.eventId === values.changedEvents[i]);
+        let originalEvent = values.events.find(event => event.id === values.changedEvents[i]);
+
+        if (!!eventChanges || !! originalEvent) {
```

Comment:
`find` returns `undefined` if there is no element in the array that satisfies the testing function. So, the check should be working. Of course, correct me if I'm wrong.

Reply:
If `values.editedEvents.find` has no element in the array, which satisfies the test, `eventChanges` is `undefined` and `!!eventChanges` is false (`!undefined` is `true` and `!!undefined` is `false`). Same for `!!originalEvent`. So the only case where `(!!eventChanges || !! originalEvent)` is **not** true, is if **both** `eventChanges` and `originalEvent` are `undefined`. Therefore the code inside the if is always executed, if either of the `find` executions finds a corresponding event and only not executed, if both of them don't find the event. 
What I thought was probably meant was `(!eventChanges || !originalEvent)`, so that the error is displayed, if either `values.editedEvents.find` or `values.events.find` or both don't find the event, and no error is displayed, if the event is found in both cases.

Reply:
oh, now I get what you wanted... you're right, should be `!` instead of `!!`